web/src/App.tsx

```
@@ -110,6 +110,17 @@ const AppContent = () => {
   useGlobalShortcut(['Control', 'k'], openCommandPalette);
   useGlobalShortcut(['Meta', 'k'], openCommandPalette);
 
+  // --- Toast Limiter ---
+  const { toasts } = useToasterStore();
+  const MAX_TOASTS = 3;
+
+  useEffect(() => {
+    toasts
+      .filter((t) => t.visible) // Only consider visible toasts
+      .filter((_, i) => i >= MAX_TOASTS) // Get toasts beyond the limit
+      .forEach((t) => toast.dismiss(t.id)); // Dismiss them
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)